libs-scala/scala-utils/src/main/scala/com/daml/scalautil/nonempty/NonEmpty.scala

```diff
@@ -54,6 +54,11 @@ sealed abstract class NonEmptyColl {
     unsafeNarrow(bb.result())
   }
 
+  final def mk[Fct, A, C[X] <: imm.Iterable[X]](into: Fct, hd: A, tl: A*)(implicit
```

Comment:
Minor: How about adding a comment to explain how `mk` differs from `apply` and when it should be used? I wouldn't be able to figure out from the type signature alone when it's better to use `apply` or `mk`.